### PR TITLE
Better ValidatedPhysicsInit errors

### PR DESCRIPTION
**Pull request by Kefta**
Should now clarify why PhysicsInit failed. Partially fixes https://github.com/FPtje/DarkRP/issues/2861.

### CODE REVIEW

File: gamemode/modules/base/sh_util.lua [hunk 1]
```diff
@@ -252,24 +252,92 @@ end
...
+                        if file.Exists(mdl, "GAME") then
+                            mdlstate = 1
+                        else
+                            mdlstate = 0
+                        end
+
+                        mdlcache[mdl] = mdlstate
+
+                        -- If our state has changed, refresh the error
+                        goto ModelError
```
Comment (FPtje) [inline gamemode/modules/base/sh_util.lua]: Hold up, this makes the code really hard to read

File: gamemode/modules/base/sh_util.lua [hunk 2]
```diff
@@ -252,24 +252,92 @@ end
...
+            -- Cache missing model states to prevent repetitive disk accesses
+            -- 0 = no model file
+            -- 1 = no model physics
+            -- 2 = has model physics, hit physics limit
+            local mdlstate = mdlcache[mdl]
+            local checkmdl = mdlstate == nil
+
+            if checkmdl then
+                if file.Exists(mdl, "GAME") then
+                    if util.IsValidProp(mdl) then
```
Comment (FPtje) [inline gamemode/modules/base/sh_util.lua]: 5 nested if-statements is too much. It makes it hard to follow the code.

File: gamemode/modules/base/sh_util.lua [hunk 3]
```diff
@@ -252,24 +252,92 @@ end
...
+                err = string.format("%s has missing model \"%s\" and will be invisible and motionless", ent, mdl)
+                errHints = {
+                    "Did you type the model name correctly?",
+                    "Is the model from an addon that is not installed on the server?",
+                    "Is the model from a game that isn't properly mounted? E.g. Counter Strike: Source"
+                }
+            elseif mdlstate == 1 then
+                err = string.format("%s has model \"%s\" with no physics and will be motionless", ent, mdl)
+                errHints = {
+                    "Are you using a viewmodel instead of a worldmodel?",
```
Comment (FPtje) [inline gamemode/modules/base/sh_util.lua]: This hint is very specific for weapons. It would be confusing if you were to get this hint for e.g. spawned money.

Comment (Kefta) [reply gamemode/modules/base/sh_util.lua]: I'll just move the hint down to give priority to the more common case (no physics model).

**Comment by Kefta** [comment]
Simplified the case code which naturally reduced the nesting and removed the need for a goto.

**Comment by FPtje** [comment]
See https://github.com/FPtje/DarkRP/commit/6bbbebe46614e6f48d7d92f6a46603030090afe3
